src/Microsoft.DotNet.Build.CloudTestTasks/UploadToAzure.cs

```diff
@@ -91,31 +91,51 @@ public override bool Execute()
             }
 
             // first check what blobs are present
-            string checkListUrl = $"{AzureHelper.GetContainerRestUrl(AccountName, ContainerName)}?restype=container&comp=list"; 
+            string checkListUrl = $"{AzureHelper.GetContainerRestUrl(AccountName, ContainerName)}?restype=container&comp=list&maxresults=3000";
 
             HashSet<string> blobsPresent = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
             try
             {
                 using (HttpClient client = new HttpClient())
                 {
+                    string nextMarker = string.Empty;
                     var createRequest = AzureHelper.RequestMessage("GET", checkListUrl, AccountName, AccountKey);
 
-                    Log.LogMessage(MessageImportance.Low, "Sending request to check whether Container blobs exist");
+                    Log.LogMessage(MessageImportance.Low, "Sending request(s) to enumerate existing blobs");
                     using (HttpResponseMessage response = await AzureHelper.RequestWithRetry(Log, client, createRequest))
                     {
                         var doc = new XmlDocument();
-                        doc.LoadXml(await response.Content.ReadAsStringAsync());
-
+                        string rawXml = await response.Content.ReadAsStringAsync();
+                        doc.LoadXml(rawXml);
                         XmlNodeList nodes = doc.DocumentElement.GetElementsByTagName("Blob");
 
                         foreach (XmlNode node in nodes)
                         {
                             blobsPresent.Add(node["Name"].InnerText);
                         }
 
-                        Log.LogMessage(MessageImportance.Low, "Received response to check whether Container blobs exist");
+                        nextMarker = doc.DocumentElement.GetElementsByTagName("NextMarker").Cast<XmlNode>().FirstOrDefault()?.InnerText;
```

Comment:
\<nit> I noticed that `GetElementsByTagName` looks for descendants rather than children, which could affect perf, but I bet it doesn't matter.

Reply:
correct.  The implementation in DownloadFromAzure is probably slightly more performant but we're talking on the range of a few milliseconds.  Idea was to just not touch anything except adding marker support .